packages/@aws-cdk/pipelines/lib/legacy/actions/publish-assets-action.ts

```diff
@@ -127,6 +135,7 @@ export class PublishAssetsAction extends CoreConstruct implements codepipeline.I
         },
       },
     });
+    this.buildSpec = props.buildSpec ? codebuild.mergeBuildSpecs(props.buildSpec, buildSpec) : buildSpec;
```

Comment:
So the idea is that the user can augment the existing build spec, but not override anything. Is that it?

Reply:
Not entirely. Some things may be overwritten, but in general they will be merged.

Reply:
Yes, that's true. Objects and arrays will be merged. But for primitives (let's say, `run-as`), the one generated by the action has precedence over the one provided by the user.